### PR TITLE
 Control package status and version

### PR DESCRIPTION
**Pull request by Wiston999**
#### Pull Request (PR) description

Parametrize package ensure value. This lets one install an specific version of the squid package. It also lets absentize or purge the package.

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -3,7 +3,7 @@
 class squid::install  {
 
   package{$::squid::package_name:
-    ensure => present,
+    ensure => $::squid::package_ensure,
```
Comment (Dan33l) [inline manifests/install.pp]: Leading colons are not required anymore now.

File: manifests/init.pp [hunk 2]
```diff
@@ -10,6 +10,7 @@
   String            $ensure_service                   = $squid::params::ensure_service,
   Pattern[/\d+ KB/] $maximum_object_size_in_memory    = $squid::params::maximum_object_size_in_memory,
   String            $package_name                     = $squid::params::package_name,
+  String            $package_ensure                   = $squid::params::package_ensure,
```
Comment (Dan33l) [inline manifests/init.pp]: more accurate `Variant[Enum['present','latest'], Pattern[/^\d.*/]]`

File: manifests/params.pp [hunk 3]
```diff
@@ -38,6 +38,7 @@
   $logformat                     = undef
   $error_directory               = undef
   $err_page_stylesheet           = undef
+  $package_ensure                = present
```
Comment (Dan33l) [inline manifests/params.pp]: Missing quotes

File: manifests/init.pp (hunk 2 above)
Comment (Wiston999) [reply manifests/init.pp]: I've created a custom type to cover all enum cases (present, latest, absent and purged) and also the Pattern case.

**Comment by bastelfreak** [comment]
Hi @Wiston999, thanks for the PR!

**Pull request merged**